Codeinwp/optimole-wp
# Fix AMP compatibility be deferring is_amp_endpoint() check until after wp action

### All Submissions:

 

### Changes proposed in this Pull Request:
 
The lazy-loading functionality is currently not working properly in AMP. This is because it is calling `is_amp_endpoint()` as early as `after_setup_theme`. This will cause a `_doing_it_wrong()` warning. This function can't be invoked before the `wp` action because it needs the `WP_Query` to be instantiated. The fix is simply to defer checks to `is_amp_endpoint()` until it the page is being rendered.

### How to test the changes in this Pull Request:

1. Enable AMP plugin in Standard mode.
2. Ensure no validation errors from Optimole are reported.
3. Ensure error log is free of notices.

### Other information:
